cpp/src/parquet/arrow/path_internal_test.cc

```diff
@@ -113,9 +114,50 @@ class MultipathLevelBuilderTest : public testing::Test {
       ArrowWriteContext(default_memory_pool(), arrow_properties_.get());
 };
 
-TEST_F(MultipathLevelBuilderTest, NonNullableSingleListNonNullableEntries) {
+class MultipathLevelListTool {
+ public:
+  static std::shared_ptr<::arrow::DataType> get_list(
+      const std::shared_ptr<::arrow::Field>& element) {
+    return ::arrow::list(element);
+  }
+  static std::shared_ptr<::arrow::DataType> get_list(
+      const std::shared_ptr<::arrow::DataType>& element) {
+    return ::arrow::list(element);
+  }
+  static std::shared_ptr<::arrow::DataType> get_large_list(
+      const std::shared_ptr<::arrow::Field>& element) {
+    return ::arrow::large_list(element);
+  }
+};
+
+class MultipathLevelListViewTool {
+ public:
+  static std::shared_ptr<::arrow::DataType> get_list(
+      const std::shared_ptr<::arrow::Field>& element) {
+    return ::arrow::list_view(element);
+  }
+  static std::shared_ptr<::arrow::DataType> get_list(
+      const std::shared_ptr<::arrow::DataType>& element) {
+    return ::arrow::list_view(element);
+  }
+  static std::shared_ptr<::arrow::DataType> get_large_list(
+      const std::shared_ptr<::arrow::Field>& element) {
+    return ::arrow::large_list_view(element);
+  }
+};
+
+template <typename ListTool>
```

Comment:
You don't need to template this by type, since all factories have the same signature anyway.